gradle-baseline-java/src/main/java/com/palantir/baseline/plugins/BaselineTesting.java

```diff
@@ -64,88 +56,52 @@ public void apply(Project project) {
             }
         });
 
-        project.getPluginManager().withPlugin("java-base", unusedPlugin -> {
+        project.getPluginManager().withPlugin("java", unusedPlugin -> {
+            JavaPluginExtension javaPluginExtension = project.getExtensions().getByType(JavaPluginExtension.class);
+            TestingExtension testingExtension = project.getExtensions().getByType(TestingExtension.class);
+
             TaskProvider<CheckJUnitDependencies> checkJUnitDependencies =
                     project.getTasks().register("checkJUnitDependencies", CheckJUnitDependencies.class);
 
-            project.getExtensions()
-                    .getByType(JavaPluginExtension.class)
-                    .getSourceSets()
-                    .configureEach(sourceSet -> {
-                        getTestTaskForSourceSet(project, sourceSet).ifPresent(testTask -> {
-                            testTask.dependsOn(checkJUnitDependencies);
-                        });
-
-                        ifHasResolvedCompileDependenciesMatching(
-                                project,
-                                sourceSet,
-                                BaselineTesting::requiresJunitPlatform,
-                                () -> fixSourceSet(project, sourceSet));
+            testingExtension
+                    .getSuites()
+                    .named(JvmTestSuitePlugin.DEFAULT_TEST_SUITE_NAME, JvmTestSuite.class, testSuite -> {
+                        testSuite.useJUnitJupiter();
```

Comment:
For backwards compatibility reasons, Gradle uses the legacy JUnit4 test toolchain for the default `test` test suite and the JUnit Platform test toolchain for everything else.

https://github.com/gradle/gradle/blob/4bb4af6421896c3ef094349be8e33098d343db2b/platforms/jvm/plugins-jvm-test-suite/src/main/java/org/gradle/api/plugins/jvm/internal/DefaultJvmTestSuite.java#L91-L100

Here we configure the default `test` test suite to also use the JUnit Platform test toolchain.